src/System.Runtime.Serialization.Xml/tests/DataContractSerializer.cs

```diff
@@ -3055,6 +3054,2031 @@ public static void DCS_BasicRoundtripDCRDefaultCollections()
         SerializationTestTypes.ComparisonHelper.CompareRecursively(defaultCollections, actual);
     }
 
+    [Fact]
+    public static void DCS_BasicPerSerializerRoundTripAndCompare_IObjectRef()
+    {
+        var setting = new DataContractSerializerSettings()
+        {
+            DataContractResolver = new SerializationTestTypes.SimpleResolver()
+        };
+
+        //DCExplicitInterfaceIObjRef
+        string baseline1 = @"<ObjectContainer xmlns=""http://schemas.datacontract.org/2004/07/SerializationTestTypes"" xmlns:i=""http://www.w3.org/2001/XMLSchema-instance""><_data i:type=""a:SerializationTestTypes.DCExplicitInterfaceIObjRef***"" xmlns:a=""http://schemas.datacontract.org/2004/07/SerializationTestTypes.DCExplicitInterfaceIObjRef***""><data z:Id=""i1"" xmlns:z=""http://schemas.microsoft.com/2003/10/Serialization/""><Data z:Ref=""i1""/></data></_data><_data2 i:type=""a:SerializationTestTypes.DCExplicitInterfaceIObjRef***"" xmlns:a=""http://schemas.datacontract.org/2004/07/SerializationTestTypes.DCExplicitInterfaceIObjRef***""><data z:Ref=""i1"" xmlns:z=""http://schemas.microsoft.com/2003/10/Serialization/""/></_data2></ObjectContainer>";
```

Comment:
These test cases are very similar. It'd be better to refactor the pattern into a method, e.g.

```c#
        private static void TestObjectInObjectContainerWithSimpleResolver<T>(T o, string baseline) 
        {
            var setting = new DataContractSerializerSettings()
            {
                DataContractResolver = new SerializationTestTypes.SimpleResolver()
            };

            var value = new SerializationTestTypes.ObjectContainer(o);
            var actual = SerializeAndDeserialize(value, baseline, setting);
            SerializationTestTypes.ComparisonHelper.CompareRecursively(value, actual);
        }
```
And we can call the method like,

```c#
TestObjectInObjectContainerWithSimpleResolver(new SerializationTestTypes.DCExplicitInterfaceIObjRef(true),
    "baseline string....");
```